src/CoreWCF.Primitives/src/CoreWCF/Telemetry/WcfInstrumentationOptions.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using CoreWCF.Channels;
+
+namespace CoreWCF.Telemetry;
+
+
+/// <summary>
+/// Options for WCF instrumentation.
+/// </summary>
+public class WcfInstrumentationOptions
+{
+    /// <summary>
+    /// Gets or sets an action to enrich an Activity.
+    /// </summary>
+    /// <remarks>
+    /// <para><see cref="Activity"/>: the activity being enriched.</para>
+    /// <para>string: the name of the event. Will be one of the constants in <see cref="WcfEnrichEventNames"/>.
+    /// </para>
+    /// <para>object: the raw <see cref="Message"/> from which additional information can be extracted to enrich the activity.
+    /// </para>
+    /// </remarks>
+    public Action<Activity, string, object>? Enrich { get; set; }
+
+    /// <summary>
+    /// Gets or sets a Filter function to filter instrumentation for requests on a per request basis.
+    /// The Filter gets the Message, and should return a boolean.
+    /// If Filter returns true, the request is collected.
+    /// If Filter returns false or throw exception, the request is filtered out.
+    /// </summary>
+    public Func<Message, bool>? IncomingRequestFilter { get; set; }
```

Comment:
We don't do this for other scenarios - filtering is typically enabled after OTel by something like the OTel collector.